### ISSUE TITLE
Supporting the first bottom solid layer

### ISSUE THREAD

**Issue by xxsteven69xx** [slic3r/Slic3r#4778]
Let me start off saying I have no experience programming. I wonder how hard it would be to implement a way to guarantee that the first bottom solid layer will allways attach to infill/perimeter.

Example of problem:

![Start of first bottom layer](https://user-images.githubusercontent.com/20728844/55816138-366ae180-5ab7-11e9-911c-58fe03055dae.PNG)

Idea 1 is to take the perimeter and grow/expand it until something like (first_bottom_solid_layer_contact_area)=(first_bottom_solid_layer_perimeter_length) x 50%

![Fist bottom layer idea 1](https://user-images.githubusercontent.com/20728844/55816427-da548d00-5ab7-11e9-9906-dfdbc6eacd33.jpg)

Idea 2 is to just make sure it is attached on all ends

![Fist bottom layer idea 2](https://user-images.githubusercontent.com/20728844/55816494-f821f200-5ab7-11e9-9e14-79a276cd7b5c.jpg)

**Comment by supermerill** [comment]
Somethign like that?
![support](https://user-images.githubusercontent.com/6536403/55820784-37723200-5afc-11e9-9706-7314d010ab46.jpg)

I'm doing something very similar in my fork in https://github.com/supermerill/Slic3r/wiki/Dense-infill-under-solid-layer

> how hard

very

**Comment by xxsteven69xx** [comment]
Yeah, that would work. Just trying brain storm on how to make slic3r more idiot proof. I'm a fan of less infill and more perimeters. It has gotten me into trouble a few times. 

![IMG_20190406_210725](https://user-images.githubusercontent.com/20728844/55822174-9f0c8b00-5ac4-11e9-8564-e51fb12c5aa4.jpg)
![IMG_20190406_210643](https://user-images.githubusercontent.com/20728844/55822175-9f0c8b00-5ac4-11e9-8f03-9a077e64fbb4.jpg)
